src/pattern.rs

```diff
@@ -147,18 +260,30 @@ impl PatternLayout {
     }
 
     /// Writes the specified `LogRecord` to the specified `Write`r according
-    /// to its pattern.
+    /// to its pattern. This method should *not* be used for console `Write`rs.
     pub fn append<W>(&self, w: &mut W, record: &LogRecord) -> io::Result<()> where W: Write {
         let location = Location {
             module_path: record.location().module_path(),
             file: record.location().file(),
             line: record.location().line(),
         };
-        self.append_inner(w, record.level(), record.target(), &location, record.args())
+        self.append_inner(w, false, record.level(), record.target(), &location, record.args())
+    }
+
+    /// Writes the specified `LogRecord` to the specified `Write`r according
+    /// to its pattern. This method should be used for console `Write`rs.
+    pub fn append_console<W>(&self, w: &mut W, record: &LogRecord) -> io::Result<()> where W: Write {
+        let location = Location {
+            module_path: record.location().module_path(),
+            file: record.location().file(),
+            line: record.location().line(),
+        };
+        self.append_inner(w, true, record.level(), record.target(), &location, record.args())
     }
 
     fn append_inner<W>(&self,
```

Comment:
It seems weird to pass a writer to this method but it internally pulls a `term::stdout()` as well. Maybe this method should take in a `W: Terminal` instead?